File: app/src/main/java/com/github/aakumykov/sync_dir_to_cloud/interfaces/iSyncTaskStopper.java
package com.github.aakumykov.sync_dir_to_cloud.interfaces;

import com.github.aakumykov.sync_dir_to_cloud.domain.entities.SyncTask;

public interface iSyncTaskStopper {

    void stopSyncTask(final SyncTask syncTask, final Callbacks callbacks);

    interface Callbacks {
        void onSyncTaskStopped();
        void onSyncTaskStoppingError(Exception e);
    }
}

File: app/src/main/java/com/github/aakumykov/sync_dir_to_cloud/interfaces/iSyncTaskStopper.kt
package com.github.aakumykov.sync_dir_to_cloud.interfaces


import com.github.aakumykov.sync_dir_to_cloud.domain.entities.SyncTask

interface iSyncTaskStopper {

    fun stopSyncTask(syncTask: SyncTask, callbacks: Callbacks)

    interface Callbacks {
        fun onSyncTaskStopped()
        fun onSyncTaskStoppingError(e: Exception)
    }
}
